### DISCUSSION TITLE
Is Spectre adapted for Domain specific language prompt app ?

### DISCUSSION THREAD

**Discussion by SadE54** [spectreconsole/spectre.console#990] [General]
Hi ,

I have to develop an application based on a prompt asking for DSL  commands using a prompt
An external parser is used , but I would like to  propose syntax highlighting and autocompletion (form list of keywords).
I looked at terminal.gui , But the colors range seems limited.
Then I found Spectre , but I don't know if it's possible to implement easily such features , and mostly if it's adapted for such application 🤔

**Comment by phil-scott-78** [comment]
Sounds like you might want to take a look at the [Radline](https://github.com/spectreconsole/radline) library. It's a work in progress, but I know @jonsequitur has had some luck using it with his [dotnet-repl project](https://github.com/jonsequitur/dotnet-repl).
